### PR TITLE
Address tpx convergence issue

### LINKED ISSUE #605: Setting T and P for PureFluid heptane fails at low temperatures
### Cantera version
2.4.0 and 2.5.0a2

### Steps to reproduce
Run the following in Python:

```py
p = ct.Heptane()
p.TP = 204.4, ct.one_atm
```
Generates the exception:
```
CanteraError thrown by Substance::set_TPp:
no convergence for P* = 6.19745e-07, V* = 0.257196
```
Setting the state to a temperature of 204.5 or higher works fine. The minimum temperature for the model is supposed to be 182.56 K.

### PR DESCRIPTION
**Pull request by ischoegl**
**Changes proposed in this pull request**

- Fix `tpx::Substance::set_TPp` so that it converges on the nearly incompressible liquid branch at low temperature. There, the saturation pressure becomes so small that the solver's requested *relative* pressure tolerance (`ErrP * Pressure`) is finer than one floating-point ULP of pressure, so the iteration could never satisfy it and bailed out with a spurious `No convergence` error. The iteration now (a) detects a sign change of the pressure residual between successive iterates to bracket the root, and (b) terminates by accepting the root once the volume step drops below the floating-point resolution of the specific volume. `TP` setters now work down to each pure fluid's minimum temperature.
- Make `tpx::Substance::Tsat` reject pressures below the minimum pressure (evaluated as the saturation pressure at the minimum temperature) with the meaningful `Illegal pressure value` error, instead of failing later with the opaque `No convergence`. This removes a deferred `@todo`.
- Add a regression test (`test_set_TP_near_min_temp`) that sets a compressed-liquid state at the minimum temperature for every pure fluid, and re-enable the previously disabled below-triple-point `Tsat` test.

**If applicable, fill in the issue number this pull request is fixing**

Closes #605

**If applicable, provide an example illustrating new features this pull request is introducing**

The example from the issue now succeeds instead of raising `CanteraError`:

```python
import cantera as ct

p = ct.Heptane()
p.TP = 204.4, ct.one_atm      # previously: "no convergence for P* = ..."
print(p.density)              # 777.37 kg/m^3

# also works all the way down to the minimum temperature
p.TP = p.min_temp, ct.one_atm
print(p.T, p.density)         # 182.56  803.57 kg/m^3
```

AI Statement (required)

- **Extensive use of generative AI.** The root-cause analysis and implementation were carried out with Claude Code (Opus 4.8). All scoping and design decisions were made by the contributor, and every generated change was reviewed, understood, and validated (numerical reproduction of the solver behavior plus the full pure-fluid and units test suites) by the contributor.

Checklist

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (scons build & scons test) and unit tests address code coverage
- [x] Style & formatting of contributed code follows contributing guidelines
- [x] AI Statement is included
- [x] The pull request is ready for review

### CODE REVIEW

File: src/tpx/Sub.cpp [hunk 1]
```diff
@@ -801,9 +800,44 @@ void Substance::set_TPp(double Temp, double Pressure)
...
+        double res = P_here - Pressure;
+        if (have_prev && kbr == 0 && res*res_prev < 0.0) {
+            // The residual changed sign between the previous and current
+            // specific volume, so the root lies between them. On the steep
+            // liquid branch at low temperature the iteration can step across
+            // the (very thin) target band without the logic below ever
+            // recording a bracket, leaving the solver unable to converge
+            // (GitHub issue #605). Capture the bracket explicitly so the
+            // in-bracket refinement below can proceed. Pressure decreases with
+            // specific volume here, so the smaller volume is the lower bound.
```
Comment (speth) [inline src/tpx/Sub.cpp]: I've noticed Claude likes to reference GitHub Issues in its comments, but I think we should keep references to now-solved issues out of the code; a future reader should only be presented with an explanation of what the code does now, not how it's a fix for some bug that used to exist. Likewise for the other comment block. Keeping the reference in the new regression test is reasonable, though.

**Review by speth — CHANGES_REQUESTED** [review]
Thanks, @ischoegl. Just one minor suggestion.

**Comment by ischoegl** [comment]
> Thanks, @ischoegl. Just one minor suggestion.

Sure! Taken care of.

**Review by speth — APPROVED** [review]
Thanks, @ischoegl. This looks good to me.
